include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -580,6 +574,22 @@ auto mode(RandomAccessIterator first, RandomAccessIterator last) -> std::vector<
     return modes;
 }
 
+template<class RandomAccessContainer>
+inline auto sorted_mode(RandomAccessContainer & v)
+{
+    return sorted_mode(v.begin(), v.end());
+}
+
+template<class RandomAccessIterator>
+auto mode(RandomAccessIterator first, RandomAccessIterator last)
+{
+    using Z = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    std::vector<Z> sorted(first, last);
+    std::sort(sorted.begin(), sorted.end());
```

Comment:
I still recommend against copying, and instead recommend documenting that `mode` mutates its inputs.